great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -4721,8 +4721,83 @@ def _initialize_usage_statistics(
             data_context=self,
             data_context_id=self._data_context_id,
             usage_statistics_url=usage_statistics_config.usage_statistics_url,
+            oss_id=self._get_oss_id(),
         )
 
+    @classmethod
+    def _get_oss_id(cls) -> uuid.UUID | None:
```

Comment:
All of this usage stats logic in the `DataContext` could and probably should be moved to a separate class.

Maybe a future refactor once this work gets done.